### ISSUE TITLE
Lerna exec hangs

### ISSUE THREAD

**Issue by UnderKoen** [lerna/lerna#3875]
## Current Behavior

When executing `lerna exec echo 0` on windows it does not exit and keeps hanging.

## Expected Behavior

It exiting after execution.

## Steps to Reproduce

1. execute `lerna exec echo 0`
2. wait
3. wait
4. wait
5. :/

This issue may not be prioritized if details are not provided to help us reproduce the issue.

### Failure Logs / Configuration

<details><summary>lerna.json</summary><p>

```json
{
  "$schema": "node_modules/lerna/schemas/lerna-schema.json",
  "version": "0.0.0"
}
```

</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
<!-- If you have a `lerna-debug.log` available, please paste it here -->
<!-- Otherwise, feel free to delete this <details> block -->
```

</p></details>

### Environment

  System:
    OS: Windows 10 10.0.19045
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
  Binaries:
    Node: 18.16.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 9.5.1 - C:\Program Files\nodejs\npm.CMD
  Utilities:
    Git: 2.33.0.
  npmPackages:
    lerna: ^7.4.1 => 7.4.1

**Comment by fahslaj** [comment]
Hi @UnderKoen, thank you for reporting this issue. As this changed when you updated Nx, could you please provide the version of Nx you were on and the version you updated to?

**Comment by UnderKoen** [comment]
From 16.6.0 to 17.0.1

**Comment by fahslaj** [comment]
Lerna does not officially support Nx 17+ yet - we are still in the process of testing Lerna & Nx 17+. For now, I would suggest sticking to Nx < 17. We will have a new version of Lerna out very soon with Nx 17+ support. I'll follow up here when it is released.

**Comment by yaozhenx** [comment]
Aren't lerna and nx from the same company / team? You do not do any testing with the latest version of nx while releasing a new version of lerna (or vice versa for the nx team)?

**Comment by fahslaj** [comment]
Yes, we maintain both Nx and Lerna and are testing them together. We will release a new version of Lerna soon with support for the newest Nx.

I am trying to reproduce this issue with the environment provided above, and I am unable to cause `lerna exec echo 0` to hang as described. This is on windows with node 18.16.0, npm 9.5.1, lerna 7.4.1, and Nx 17.0.3. Does this hanging behavior happen to you for all Lerna repos or just one? Can you please provide a link to a repo that consistently causes this problem?

**Comment by UnderKoen** [comment]
https://github.com/UnderKoen/lerna-hangs-repro

**Comment by yaozhenx** [comment]
Any update?

**Comment by yaozhenx** [comment]
A friendly gentle ping...
